### DISCUSSION TITLE
How to migrate naming convention?

### DISCUSSION THREAD

**Discussion by jiyolla** [sqlalchemy/alembic#906] [Usage Questions] [answered]
https://github.com/sqlalchemy/sqlalchemy/discussions/7008

The same post in sqlalchemy.

Will close both when solved in either post.

question copied by mike:

Up until now, I managed the database with anonymous constraints.

And recently, I ran into situation when two foreign keys exist in the same table and alembic need a name to work correctly.

So giving a naming convention seems the most recommended solution, but I think introducing the 'new rules' into existing database would definitively cause some troubles.

Some constraints are not named as rules suggest, and alembic wouldn't find them correctly.

Is there a correct procedure to introduce the naming convention amid?

Should I just manually write a script to rename the legacy constraints?

**Comment by zzzeek** [comment, marked as answer]
if you have a whole DB with anonymously named constraints and you now have two FKs that need names, I would not advise going with a naming convention unless you really wanted to get into that.   for now, give these specific FKs an explicit name so that you can manage them directly without impacting your whole database schema.

**Reply by zzzeek** [reply]
if you wanted to migrate the whole schema, you'd likely need to write these rules manually based on inspecting the current DB, getting all the FK and other constraint names, and matching them up to the expected convention.  you'd need a staging or test database and you'd likely need a lot of iterative development to get it to work correctly.  also im not sure if your target database supports renaming of constraints , if any of these renames involve dropping and recreating, there could be more complexities implied by that.  for any production-level DB with a lot of data that you can't easily just recreate, I'd advise against getting into it.

**Reply by jiyolla** [reply]
Thank you!
That was very helpful.

My database is rather small now so that manually writing scripts would still be manageable. I just wanted to know the best practice.

**Reply by jiyolla** [reply]
Is there any reason not to use a sqlalchemy level naming convention even when starting from stratch?

**Reply by zzzeek** [reply]
you should use the naming convention right from the start.   changing them / adding them is not really very easy.

**Reply by jiyolla** [reply]
Is there any reason not to enforce a sqlalchemy level naming convention even from start?

As the default behavior is not having them.

I mean if there's no cons, wouldn't having a default naming convention be better?

**Reply by zzzeek** [reply]
sqlalchemy wouldnt want to invent one (even though we have a slight one present for index=True already) and we also wouldnt want to disallow using the naming conventions implemented by the backend database, as they are also quite popular.
